### ISSUE TITLE
Old maximum grpcio version is blocking other dependencies

### ISSUE THREAD

**Issue by christophbergemann** [skypilot-org/skypilot#1956]
skypilot requires grpcio<=1.43.0, which is quite old. This is causing unresolvable version conflicts in some cases. While my example (below) may be quite a particular use case, I think this is going to be a more general problem.

My example is using

```
pymilvus>=2.2.8
skypilot>=0.2.5
```

leading to

```
The conflict is caused by:
    pymilvus 2.2.8 depends on grpcio<=1.53.0 and >=1.49.1
    skypilot 0.2.5 depends on grpcio<=1.43.0 and >=1.32.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict
```

**Comment by concretevitamin** [comment]
Thanks for the report @christophbergemann! What's your Python version & OS?

@Michaelvll: wondering if we can do https://github.com/ray-project/ray/blob/72e5f6be51808a77fe15a87b206d8214fc91abda/python/requirements.txt#L25-L30.

**Comment by christophbergemann** [comment]
Running on Windows 11 with Python 3.10.

**Comment by Michaelvll** [comment]
Good point! It seems the problem has been solved by the upstream `ray>=2.1`. Since we currently allow lower `ray` version, we are blocked by the #1734.
https://github.com/ray-project/ray/blob/ray-2.1.0/python/setup.py

**Comment by concretevitamin** [comment]
This has now been relaxed in 0.3: 

https://github.com/skypilot-org/skypilot/blob/3deccb2a7360d086952601188381c28f932e118e/sky/setup_files/setup.py#L97-L101

Please feel free to reopen this if it doesn't fix the issue for you.
